plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -118,11 +118,17 @@ def check_all_and_associate(wanted, conduit, download_deferred, catalog):
         # fields = model.unit_key_fields + ('_storage_path',)
         unit_generator = (model(**unit_tuple._asdict()) for unit_tuple in values.copy())
         for unit in units_controller.find_units(unit_generator):
-            # Existing RPMs, DRPMs and SRPMs are disqualified when the associated
-            # package file does not exist and downloading is not deferred.
-            if not download_deferred and unit_type in (
-                    ids.TYPE_ID_RPM, ids.TYPE_ID_SRPM, ids.TYPE_ID_DRPM):
-                if unit._storage_path is None or not os.path.isfile(unit._storage_path):
+            package_type = unit_type in (ids.TYPE_ID_RPM, ids.TYPE_ID_SRPM, ids.TYPE_ID_DRPM)
+            valid_file = unit._storage_path is not None and os.path.isfile(unit._storage_path)
```

Comment:
Continuing to pick on naming, I suggest calling this something more explicit like ``file_exists``. "valid" is a vague term that could mean all kinds of things, but really in this case I think we're just confirming if it exists. That criteria is simple enough to just say it in the variable name.